src/utils/addresses.ts

```diff
@@ -29,3 +30,83 @@ export const copyAddressToClipboard = (addressHash: AddressHash) => {
   Clipboard.setString(addressHash)
   Toast.show('Address copied!')
 }
+
+export const fetchAddressesData = async (addressHashes: AddressHash[]) => {
```

Comment:
Discussion: I'm not sure about calling apis from a file in the `utils` folder. For me, those utilities only execute local code, without side effects. Question: where could this be stored instead?

Reply:
That's a good point, I also had doubts before moving it to utils and I am glad you are bringing it up. I'll try to think of a new proposal.

Reply:
How about moving it inside the `src/api/client.ts` file, similarly to `buildSweepTransactions` and `signAndSendTransaction`?

Reply:
This is what I propose:

<details><summary>See diff</summary>

```diff
diff --git a/src/api/client.ts b/src/api/client.ts
index 1028621..66dba47 100644
--- a/src/api/client.ts
+++ b/src/api/client.ts
@@ -53,6 +53,45 @@ export class Client {
 
     return data
   }
+
+  async fetchAddressesData(addressHashes: AddressHash[]) {
+    const results = []
+
+    for (const addressHash of addressHashes) {
+      console.log('⬇️ Fetching address details: ', addressHash)
+      const { data } = await this.explorerClient.getAddressDetails(addressHash)
+      const availableBalance = data.balance
+        ? data.lockedBalance
+          ? (BigInt(data.balance) - BigInt(data.lockedBalance)).toString()
+          : data.balance
+        : undefined
+
+      console.log('⬇️ Fetching 1st page of address confirmed transactions: ', addressHash)
+      const { data: transactions } = await this.explorerClient.getAddressTransactions(addressHash, 1)
+
+      console.log('⬇️ Fetching address tokens: ', addressHash)
+      const { data: tokenIds } = await this.explorerClient.addresses.getAddressesAddressTokens(addressHash)
+
+      const tokens = await Promise.all(
+        tokenIds.map((id) =>
+          this.explorerClient.addresses.getAddressesAddressTokensTokenIdBalance(addressHash, id).then(({ data }) => ({
+            id,
+            balances: data
+          }))
+        )
+      )
+
+      results.push({
+        hash: addressHash,
+        details: data,
+        availableBalance: availableBalance,
+        transactions,
+        tokens
+      })
+    }
+
+    return results
+  }
 }
 
 const client = new Client(defaultNetworkSettings)
diff --git a/src/store/addressesSlice.ts b/src/store/addressesSlice.ts
index 7ce6495..8a6cfe7 100644
--- a/src/store/addressesSlice.ts
+++ b/src/store/addressesSlice.ts
@@ -32,7 +32,7 @@ import { getAddressesMetadataByWalletId } from '../storage/wallets'
 import { Address, AddressHash, AddressPartial, AddressSettings } from '../types/addresses'
 import { AddressToken } from '../types/tokens'
 import { PendingTransaction } from '../types/transactions'
-import { fetchAddressesData, storeAddressSettings } from '../utils/addresses'
+import { storeAddressSettings } from '../utils/addresses'
 import { getRandomLabelColor } from '../utils/colors'
 import { mnemonicToSeed } from '../utils/crypto'
 import { extractNewTransactions, extractRemainingPendingTransactions } from '../utils/transactions'
@@ -68,7 +68,7 @@ export const addressesDataFetched = createAsyncThunk(
     dispatch(loadingStarted())
 
     const addresses = payload
-    const results = await fetchAddressesData(addresses)
+    const results = await client.fetchAddressesData(addresses)
 
     dispatch(loadingFinished())
     return results
diff --git a/src/utils/addresses.ts b/src/utils/addresses.ts
index eaf7caf..c7507f7 100644
--- a/src/utils/addresses.ts
+++ b/src/utils/addresses.ts
@@ -19,7 +19,6 @@ along with the library. If not, see <http://www.gnu.org/licenses/>.
 import * as Clipboard from 'expo-clipboard'
 import Toast from 'react-native-root-toast'
 
-import client from '../api/client'
 import { storeAddressMetadata } from '../storage/wallets'
 import { Address, AddressHash, AddressPartial, AddressSettings } from '../types/addresses'
 
@@ -31,45 +30,6 @@ export const copyAddressToClipboard = (addressHash: AddressHash) => {
   Toast.show('Address copied!')
 }
 
-export const fetchAddressesData = async (addressHashes: AddressHash[]) => {
-  const results = []
-
-  for (const addressHash of addressHashes) {
-    console.log('⬇️ Fetching address details: ', addressHash)
-    const { data } = await client.explorerClient.getAddressDetails(addressHash)
-    const availableBalance = data.balance
-      ? data.lockedBalance
-        ? (BigInt(data.balance) - BigInt(data.lockedBalance)).toString()
-        : data.balance
-      : undefined
-
-    console.log('⬇️ Fetching 1st page of address confirmed transactions: ', addressHash)
-    const { data: transactions } = await client.explorerClient.getAddressTransactions(addressHash, 1)
-
-    console.log('⬇️ Fetching address tokens: ', addressHash)
-    const { data: tokenIds } = await client.explorerClient.addresses.getAddressesAddressTokens(addressHash)
-
-    const tokens = await Promise.all(
-      tokenIds.map((id) =>
-        client.explorerClient.addresses.getAddressesAddressTokensTokenIdBalance(addressHash, id).then(({ data }) => ({
-          id,
-          balances: data
-        }))
-      )
-    )
-
-    results.push({
-      hash: addressHash,
-      details: data,
-      availableBalance: availableBalance,
-      transactions,
-      tokens
-    })
-  }
-
-  return results
-}
-
 export const findNextAvailableAddressIndex = (startIndex: number, skipIndexes: number[] = []) => {
   let nextAvailableAddressIndex = startIndex

 ```
</details>

Reply:
Putting this in `api` sounds good. 
And what about:
- moving `buildSweepTransactions ` and `signAndSendTransaction ` out of `client.ts` to `src/api/transactions.ts` (they don't really need to be attached to the client class IMO)
- moving `fetchAddressesData`to `src/api/addresses`

WDYT?

Reply:
Sounds like a great idea man, thanks!

Reply:
I took care of this in the following PR because I was working on a lot of refactoring, I hope that's ok!